hazelcast/src/main/java/com/hazelcast/config/PersistentMemoryDirectoryConfig.java

```diff
@@ -0,0 +1,158 @@
+
+    /**
+     * Creates an instance with the {@code directory} specified.
+     *
+     * @param directory The directory where the persistent memory is
+     *                  mounted at
+     * @throws NullPointerException if {@code directory} is {@code null}
+     */
+    public PersistentMemoryDirectoryConfig(@Nonnull String directory) {
+        this.directory = requireNonNull(directory);
+    }
+
+    /**
+     * Creates an instance with the {@code directory} and NUMA node specified.
+     * <p/>
+     * Note that setting {@code numaNode} to -1 on any of the configured
+     * {@link PersistentMemoryDirectoryConfig} instances disables
+     * NUMA-aware persistent memory allocation.
+     *
+     * @param directory The directory where the persistent memory is
+     *                  mounted at
+     * @param numaNode  The NUMA node that the persistent memory mounted
+     *                  to the given directory is attached to.
+     * @throws NullPointerException if {@code directory} is {@code null}
+     */
+    public PersistentMemoryDirectoryConfig(@Nonnull String directory, int numaNode) {
+        this.directory = requireNonNull(directory);
+        this.numaNode = numaNode;
+    }
+
+    /**
+     * Constructs an instance by copying the provided {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @param directoryConfig The configuration to copy
+     * @throws NullPointerException if {@code directoryConfig} is {@code null}
+     */
+    public PersistentMemoryDirectoryConfig(@Nonnull PersistentMemoryDirectoryConfig directoryConfig) {
+        this.directory = requireNonNull(directoryConfig).directory;
+        this.numaNode = directoryConfig.numaNode;
+    }
+
+    /**
+     * Returns the directory of this {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @return the directory
+     */
+    @Nonnull
+    public String getDirectory() {
+        return directory;
+    }
+
+    /**
+     * Returns the {@code directory} of this {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @param directory the directory to set
+     * @throws NullPointerException if {@code directory} is {@code null}
+     */
+    public void setDirectory(@Nonnull String directory) {
+        this.directory = requireNonNull(directory);
+    }
+
+    /**
+     * Returns the NUMA node the persistent memory mounted to the given
+     * directory is attached to.
+     *
+     * @return the NUMA node of the persistent memory
+     */
+    public int getNumaNode() {
+        return numaNode;
+    }
+
+    /**
+     * Sets the NUMA node the persistent memory mounted to the given
+     * directory is attached to.
+     * <p/>
+     * Note that setting {@code numaNode} to -1 on any of the configured
+     * {@link PersistentMemoryDirectoryConfig} instances disables
+     * NUMA-aware persistent memory allocation.
+     *
+     * @param numaNode the NUMA node to set
+     */
+    public void setNumaNode(int numaNode) {
+        this.numaNode = numaNode;
```

Comment:
What if I set `numaNode = Integer.MAX_VALUE`? Do we validate anywhere the value?

Reply:
Yes, in the PMEM malloc implementation. Such a configuration disables NUMA-aware allocations if it would be enabled otherwise.